### PR TITLE
Allow identifier to be PatientID or PatientName

### PR DESCRIPTION
**Pull request by spaiva**
- Created $identifier variable to allow parsing for PatientID or PatientName
- No longer using $lookupCenterNameUsing
- Regular Pattern of Patient ID coming from Prod file. 
- Dependencies: Dicom-Archive [PR#38](https://github.com/aces/dicom-archive-tools/pull/38)

### CODE REVIEW

File: uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm [hunk 1]
```diff
@@ -186,12 +188,21 @@ sub determineSubjectID {
...
-                            $tarchiveInfo->{'PatientName'},
-                            $tarchiveInfo->{'PatientID'},
-                            $scannerID,
-                            $this->{dbhr}
-                        );
+                                $identifier,
+                                $scannerID,
+                                $this->{dbhr}
+                            );
+
```
Comment (MounaSafiHarab) [inline uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm]: @stronnics,
this function has been updated to take 3 arguments ($identifier now replaces PatientName and PatientID), but your dicom-archive-tools pull request that updated the profileTemplate file did not update this function (it still takes in 4 arguments)

**Comment by MounaSafiHarab** [comment]
@stronnics,
is it possible to get rid of the changes related just to spaces? it is very hard to track the real changes...

**Comment by stronnics** [comment]
@MounaSafiHarab I have tried 2 times to get rid of the spaces. I believe that this is coming from Github itself. If you know a way, please let me know. I could close the PR and re-open it.

File: uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm (hunk 1 above)
Comment (stronnics) [inline uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm]: @MounaSafiHarab Thanks for the catch, made the change.

**Comment by stronnics** [comment]
I would need permissions to be able to set Labels.

File: uploadNeuroDB/NeuroDB/MRI.pm [hunk 2]
```diff
@@ -973,32 +982,32 @@ sub createNewCandID {
...
+	    if ($identifier =~ /$row->{'Alias'}/i) || ($identifier =~ /$row->{'MRI_alias'}/i);
     }
 
-    ###########################################################################	
+    ###########################################################################
     ###Get the centerID using the PSCID########################################
     ###########################################################################
 
     #extract the PSCID from $patientName
-    $subjectIDsref = getSubjectIDs($patientName,null,$dbhr);
```
Comment (MounaSafiHarab) [inline uploadNeuroDB/NeuroDB/MRI.pm]: @stronnics 

this is the only place that used to call the getSubjectIDs() within the MRI.pm, and now you replaced it with a call to this function from the prod file, do we then still need then the function in MRI.pm (around line 1001)?

File: uploadNeuroDB/imaging_upload_file.pl [hunk 3]
```diff
@@ -243,7 +241,18 @@
     print $message;
     exit 9;
 }
-$message = "\nThe uploaded file " . $uploaded_file . " has been removed\n\n";
+$message = "\nThe uploaded file " . $uploaded_file . " has been removed\n";
```
Comment (MounaSafiHarab) [inline uploadNeuroDB/imaging_upload_file.pl]: @stronnics 
I don't understand why those changes are in this pull request?

Comment (stronnics) [inline uploadNeuroDB/imaging_upload_file.pl]: I don't understand either, the missing '\n' is not highlighted as removed...

File: uploadNeuroDB/NeuroDB/MRI.pm (hunk 2 above)
Comment (stronnics) [inline uploadNeuroDB/NeuroDB/MRI.pm]: I don't think we need it in MRI.pm, but I haven't tested it yet. Will do so shortly.

Comment (cmadjar) [inline uploadNeuroDB/NeuroDB/MRI.pm]: I agree with Mouna, if the function is never used in MRI.pm, let's remove it from MRI.pm.

File: uploadNeuroDB/imaging_upload_file.pl [hunk 4]
```diff
@@ -280,6 +289,48 @@ sub getPnameUsingUploadID {
...
+  $upload_id: The Upload ID
+
+  Returns: $minc_created and $minc_inserted: count of minc created and inserted
+=cut
+
+
+sub getNumberOfMincFiles {
+
+    my $upload_id = shift;
+    my ( $minc_created, $minc_inserted, $query ) = '';
```
Comment (cmadjar) [inline uploadNeuroDB/imaging_upload_file.pl]: Why is this in this pull request?

**Comment by spaiva** [comment]
Opening new pull-request

**Pull request closed without merging**